WordPress/Classes/ViewRelated/System/WPTabBarController.m

```diff
@@ -913,7 +913,12 @@ - (void)updateNotificationBadgeVisibility
     // Discount Zendesk unread notifications when determining if we need to show the notificationsTabBarImageUnread.
     NSInteger count = [[UIApplication sharedApplication] applicationIconBadgeNumber] - [ZendeskUtils unreadNotificationsCount];
     UITabBarItem *notificationsTabBarItem = self.notificationsNavigationController.tabBarItem;
-    if (count > 0) {
+
+    NSUserDefaults *standardUserDefaults = [NSUserDefaults standardUserDefaults];
```

Comment:
What about extracting that logic to a private method? `welcomeNotificationSeen` may even be the name of the method.